prometheus/pkg/adapter/adapter.go

```diff
@@ -83,56 +85,50 @@ func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClie
 		promQL:          env.PromQL,
 		authTokenFile:   env.AuthTokenFile,
 		caCertConfigMap: env.CACertConfigMap,
+		schedule:        env.Schedule,
+		step:            env.Step,
+		lastRun:         time.Now(),
 	}
 
 	return a
 }
 
 func (a *prometheusAdapter) Start(stopCh <-chan struct{}) error {
-	completeURL := a.serverURL + apiChunkOfURL + a.promQL
-
-	var err error
-	a.req, err = http.NewRequest(`GET`, completeURL, nil)
-	if err != nil {
-		a.logger.Error("HTTP request error", zap.Error(err))
+	if err := a.readAuthTokenIfNeeded(); err != nil {
 		return err
 	}
-	if a.authTokenFile != "" {
-		content, err := ioutil.ReadFile(a.authTokenFile)
-		if err != nil {
-			a.logger.Error("Error reading authentication token from "+a.authTokenFile, zap.Error(err))
+	// pre-make an immutable HTTP Request for an instant query
+	if a.step == "" {
+		if err := a.makeHTTPRequest(); err != nil {
 			return err
 		}
-		a.req.Header.Set("Authorization", "Bearer "+string(content))
+	}
+	if err := a.makeHTTPClient(); err != nil {
+		return err
 	}
 
-	a.logger.Info(a.req)
-
-	a.client = &http.Client{}
-
-	if a.caCertConfigMap != "" {
-		caCertFile := "/etc/" + a.caCertConfigMap + "/service-ca.crt"
-		caCert, err := ioutil.ReadFile(caCertFile)
-		if err != nil {
-			a.logger.Error("Error reading CA certificate from "+caCertFile, zap.Error(err))
-			return err
-		}
-		caCertPool := x509.NewCertPool()
-		if !caCertPool.AppendCertsFromPEM(caCert) {
-			a.logger.Error("Error parsing CA certificate from " + caCertFile)
-			return err
-		}
-		a.client.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs: caCertPool,
-			},
-		}
+	sched, err := cron.ParseStandard(a.schedule)
```

Comment:
nice to align w/ the cron package.

the `CronjobSource` does similar, but also reflects the schedule in the status. see:

https://github.com/knative/eventing/blob/master/pkg/reconciler/cronjobsource/cronjobsource.go#L168-L173

Should we do the same ?

Reply:
I think that's a fine idea. Please see the next commit.